vllm/compilation/backends.py

```diff
@@ -179,9 +180,20 @@ def compile(self,
         else:
             maybe_key = \
                 f"artifact_shape_{runtime_shape}_subgraph_{graph_index}"
-        compiled_graph, handle = self.compiler.compile(
-            graph, example_inputs, additional_inductor_config, runtime_shape,
-            maybe_key)
+        if self.compilation_config.use_inductor_graph_partition:
+            partition_ops = self.compilation_config.partition_rule_ops or []
+            context = _inductor_partition_rule_context(partition_ops)
+        else:
+            context = nullcontext()
+
+        with context:
```

Comment:
Extract this into a method for easy additions for the context.
```suggestion
        with self.compile_context():
```